bin/rule.ml

```diff
@@ -88,8 +89,34 @@ let print_rule ~nd ~prelude ~md_file ~ml_files ~dirs ~root ~requires options =
       Fmt.(list ~sep:sp pp_dep) deps
       Fmt.(list ~sep:cut pp_action) (actions arg)
   in
-  pp "runtest" [];
-  if nd then pp "runtest-all" ["--non-deterministic"]
+  pp fmt "runtest" [];
+  if nd then
+    pp fmt "runtest-all" ["--non-deterministic"]
+
+let print_format_dune_rules pp_rules =
+  let open Unix in
+  let read_pipe, write_pipe = pipe () in
+  set_close_on_exec read_pipe;
+  set_close_on_exec write_pipe;
```

Comment:
`~⁠cloexec:true` on `pipe` is not working ?

Reply:
It's not available on OCaml `< 4.05` :'(

Reply:
:(